drivers/vulkan/SCsub

```diff
@@ -2,124 +2,72 @@
-    ]
     vma_sources = [thirdparty_dir + "/vk_mem_alloc.cpp"]
 
-    if env["platform"] == "windows":
-        loader_sources.append("dirent_on_windows.c")
-        env_thirdparty.AppendUnique(
-            CPPDEFINES=[
-                "VK_USE_PLATFORM_WIN32_KHR",
-                "VULKAN_NON_CMAKE_BUILD",
-                "WIN32_LEAN_AND_MEAN",
-                'API_NAME=\\"%s\\"' % "Vulkan",
-            ]
-        )
-        if not env.msvc:  # Windows 7+, missing in mingw headers
-            env_thirdparty.AppendUnique(
-                CPPDEFINES=["CM_GETIDLIST_FILTER_CLASS=0x00000200", "CM_GETIDLIST_FILTER_PRESENT=0x00000100"]
-            )
-    elif env["platform"] == "osx":
-        env_thirdparty.AppendUnique(
-            CPPDEFINES=[
-                "VK_USE_PLATFORM_MACOS_MVK",
-                "VULKAN_NON_CMAKE_BUILD",
-                'SYSCONFDIR=\\"%s\\"' % "/etc",
-                'FALLBACK_DATA_DIRS=\\"%s\\"' % "/usr/local/share:/usr/share",
-                'FALLBACK_CONFIG_DIRS=\\"%s\\"' % "/etc/xdg",
-            ]
-        )
+    if env["platform"] == "osx":
+        env.AppendUnique(CPPDEFINES=["VK_USE_PLATFORM_MACOS_MVK"])
+    elif env["platform"] == "iphone":
+        env.AppendUnique(CPPDEFINES=["VK_USE_PLATFORM_IOS_MVK"])
+    elif env["platform"] == "windows":
+        env.AppendUnique(CPPDEFINES=["VK_USE_PLATFORM_WIN32_KHR"])
     elif env["platform"] == "linuxbsd":
-        env_thirdparty.AppendUnique(
-            CPPDEFINES=[
-                "VK_USE_PLATFORM_XLIB_KHR",
-                "VULKAN_NON_CMAKE_BUILD",
-                'SYSCONFDIR=\\"%s\\"' % "/etc",
-                'FALLBACK_DATA_DIRS=\\"%s\\"' % "/usr/local/share:/usr/share",
-                'FALLBACK_CONFIG_DIRS=\\"%s\\"' % "/etc/xdg",
-            ]
-        )
-        import platform
-
-        if platform.system() == "Linux":
-            # In glibc since 2.17 and musl libc since 1.1.24. Used by loader.c.
-            env_thirdparty.AppendUnique(CPPDEFINES=["HAVE_SECURE_GETENV"])
-
-    loader_sources = [thirdparty_dir + "/loader/" + file for file in loader_sources]
-    env_thirdparty.add_source_files(thirdparty_obj, loader_sources)
-    env_thirdparty.add_source_files(thirdparty_obj, vma_sources)
-
-else:  # Always build VMA.
-    thirdparty_dir = "#thirdparty/vulkan"
-    env.Prepend(CPPPATH=[thirdparty_dir, thirdparty_dir + "/include"])
+        env.AppendUnique(CPPDEFINES=["VK_USE_PLATFORM_XLIB_KHR"])
 
-    # Build Vulkan loader library
+    # Build Vulkan memory allocator and volk
     env_thirdparty = env.Clone()
     env_thirdparty.disable_warnings()
-    vma_sources = [thirdparty_dir + "/vk_mem_alloc.cpp"]
+    if env["use_volk"]:
+        env_thirdparty.AppendUnique(CPPDEFINES=["VMA_STATIC_VULKAN_FUNCTIONS=1"])
+        vma_sources += [thirdparty_volk_dir + "/volk.c"]
 
     env_thirdparty.add_source_files(thirdparty_obj, vma_sources)
 
-
 env.drivers_sources += thirdparty_obj
 
-
 # Godot source files
-
 driver_obj = []
```

Comment:
Not very important but those lines are the style I tried to use consistently in all SCsubs.

Reply:
Fixed